### ISSUE TITLE
Feature Request: Total Blocks Broken 

### ISSUE THREAD

**Issue by AtomixCM** [PikaMug/Quests#789]
**What does your feature do?**
Instead of having to specify the exact types of blocks to be broken with corresponding values, we should be able to specify all blocks for one value.
**How would people use it?**
For the blocks mined requirement, specifying the type of block as ALL or ANY should be allowed.
This is particularly helpful for prison.

**Comment by AtomixCM** [comment]
stages:
      ordered:
        '1':
          break-block-names:
          - ANY
          break-block-amounts:
          - 10000
          break-block-durability:
          - 0
          complete-message: You have successfully broken 10000 Blocks!

**Comment by PikaMug** [comment]
Thanks for the suggestion. This could already be made possible with a Custom Objective, but I'll consider adding the functionality to the plugin itself.

**Comment by AtomixCM** [comment]
@PikaMug do you intend on adding this feature? I'm looking for any plugin that will allow this sort of quest without causing any bugs.

**Comment by PikaMug** [comment]
So I've looked into just incorporating this into the plugin and unfortunately it's not as simple as I'd hoped. Quests loads/saves blocks as an ItemStack, which has to be a specific item. In other words, you can't have an ItemStack with a name of "ANY" - it has to be something from [the Materials list](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Material.html). Rewritting those parts of the plugin to work around that would constitute an overhaul of the player data files, which I'm not willing to do.

Furthermore, it's my opinion that what you've requested doesn't really constitute a quest. Google defines the word "quest" as follows:

![quest_def](https://user-images.githubusercontent.com/2267126/60463087-0a27b280-9c19-11e9-9059-71df2b7eee35.png)

What you've requested doesn't really require any thought to accomplish. Granted, mining 10000 blocks is long, but hardly arduous as there's no pressure to hunt for a specific object. Mining X amount of blocks would better be left to a more general plugin like Jobs Reborn... or [a Quests module](https://github.com/PikaMug/Quests/wiki/Master-%E2%80%90-Custom-Quest-API).

This is all it would take to make a custom module objective that counts ANY block:

```
package me.pikamug.Blocks;

import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.block.BlockBreakEvent;

import me.blackvein.quests.CustomObjective;
import me.blackvein.quests.Quest;
import me.blackvein.quests.Quests;

public class AnyBreakBlockObjective extends CustomObjective {
	private static Quests quests = (Quests) Bukkit.getServer().getPluginManager().getPlugin("Quests");
	
	public AnyBreakBlockObjective() {
		setName("Break Blocks Objective");
		setAuthor("PikaMug");
		setShowCount(true);
		addStringPrompt("Obj Name", "Set a name for the objective", "Break ANY block");
		setCountPrompt("Set the amount of blocks to break");
		setDisplay("%Obj Name%: %count%");
	}
	
	@EventHandler(priority = EventPriority.LOW)
	public void onBlockBreak(BlockBreakEvent event) {
		Player player = event.getPlayer();
		for (Quest q : quests.getQuester(player.getUniqueId()).getCurrentQuests().keySet()) {
			incrementObjective(player, this, 1, q);
			return;
		}
	}
}
```

As you can see, this is very simple to do and would only require specifying the custom objective in the Quests Editor rather than using the built-in Break Block objective. However, as the existence of #803 implies, the module would likely want to support other Block events (Place, Use, et al.). From there, people are bound to want wildcards for Mob names, NPC IDs (see #807), delivery items, et al. and eventually for Bukkit Events which Quests doesn't inherently support (as of writing). It's a rabbit hole which would only widen with each Minecraft release - something I'm not willing to tackle myself.

**TL;DR:** Including this enhancement would be opening a can of worms. This is best left to [a module](https://github.com/PikaMug/Quests/wiki/Master-%E2%80%90-Custom-Quest-API), which I'm not willing to write. Feel free to use the code above, or enlist the help of a developer at https://www.spigotmc.org/forums/hiring-developers.55/ (which neither I nor the Quests project are associated with).

**Comment by PikaMug** [comment]
Update since I keep mentioning this: consider trying https://lectern.browsit.org/resources/resource/19/
